app/collections/favorites-button.cjsx

```diff
@@ -0,0 +1,89 @@
+
+  propTypes:
+    subject: React.PropTypes.object # a subject response from panoptes
+
+  getInitialState: ->
+    favorited: false
+
+  componentWillMount: ->
+    project_id = @props.subject.links.project.toString()
+
+    # see if the subject is in the project's favorites collection
+    # to see if it's favoriters
+    apiClient.type('projects').get(project_id)
+      .then (project) =>
+        display_name = getFavoritesName(project)
+
+        apiClient.type('collections').get({project_id, display_name}).index(0)
+          .then (favorites) =>
+            if !!favorites
+              apiClient.type('subjects').get(collection_id: favorites.id, id: @props.subject.id).index(0)
+                .then (subject) => @setState favorited: !!subject
+                .catch (e) -> throw new Error(e)
+
+  addSubjectTo: (collection) ->
+    collection.addLink('subjects', [@props.subject.id.toString()])
+      .then (collection) => @setState favorited: true
+      .catch (e) -> throw new Error(e)
+
+  removeSubjectFrom: (collection) ->
+    collection.removeLink('subjects', [@props.subject.id.toString()])
+      .then (collection) => @setState favorited: false
+      .catch (e) -> throw new Error(e)
+
+  createFavorites: (user) ->
+    project_id = @props.subject.links.project.toString()
+
+    apiClient.type('projects').get(project_id)
+      .then (project) =>
+        display_name = getFavoritesName(project)
+        project = @props.subject.links.project
+
+        owner = {id: user.id.toString(), type: "users"} #--> must supply type
+        links = {project, owner}
+        collection = {display_name, links}
+
+        apiClient.type('collections').create(collection).save()
+          .then (favorites) =>
+            @addSubjectTo(favorites)
+          .catch (e) -> throw new Error(e)
+
+  toggleFavorite: ->
+    project_id = @props.subject.links.project.toString()
+
+    auth.checkCurrent().then (user) =>
+      if user?
+        apiClient.type('projects').get(project_id)
+          .then (project) =>
+            display_name = getFavoritesName(project)
+            # check for a favorites collection in project first
+            apiClient.type('collections').get({project_id, display_name}).index(0)
+              .then (favorites) =>
+                # try to request subject from favorites collection, otherwise create it
+                if !!favorites
+                  apiClient.type('subjects').get(collection_id: favorites.id, id: @props.subject.id).index(0)
+                    .then (subject) =>
+                      if subject
+                        @removeSubjectFrom(favorites)
+                      else
+                        @addSubjectTo(favorites)
+                    .catch (e) -> throw new Error(e)
+                else
+                  @createFavorites(user)
+      else
+        alert(<div className='content-container'>You must be logged in to favorite</div>)
+
+  render: ->
+    <button
```

Comment:
Good habit to always define `type="button"` with button elements, otherwise they crazily default to type="submit"` and things get confusing if you ever end up using something in a form.